src/components/LanguageChoice/LanguagesList.tsx

```diff
@@ -1,23 +1,24 @@
 import React from "react";
 import i18n from "i18next";
 
+import { useRecoilState } from "recoil";
 import { ILanguageListProps, ILanguage } from "@src/Interfaces/ILanguage";
-import { useNavigate } from "react-router-dom";
 import { vibrateWorks } from "@src/constants/vibrateCheck";
-import { languageSelectionState } from "@src/store";
-import { useRecoilState } from "recoil";
+import { languageChangeModal, languageSelectionState } from "@src/store";
 
 export const LanguagesList = (props: ILanguageListProps) => {
-  const { languages } = props;
-  const navigate = useNavigate();
+  const { languages, navigationCallback, type } = props;
   const [, setIsLanguageChosen] = useRecoilState(languageSelectionState);
 
+  const [open, close] = useRecoilState(languageChangeModal);
+
   const handleClick = (langId: string) => {
-    (vibrateWorks) ? navigator.vibrate(100) : null;
+    vibrateWorks ? navigator.vibrate(100) : null;
     setIsLanguageChosen(langId);
     i18n.changeLanguage(langId);
     localStorage.setItem("language", JSON.stringify(langId));
-    navigate("/ZinZenFAQ");
+    if (type === "fragment" && navigationCallback) navigationCallback("/ZinZenFAQ");
+    else close(!open)
```

Comment:
@FullStackSorcerer  the core idea is to change the state of language in local storage and then use window.history.back() to close the popup. browser's state will not get updated if you will close via setState method. 
To reproduce the error ; select a language and go to journal page you will see the language popup will come again.